### PR TITLE
Add docs for taxon constraints

### PR DESCRIPTION
**Pull request by shawntanzk**
Fixes https://github.com/obophenotype/cell-ontology/issues/1701

### CODE REVIEW

File: docs/taxon-restrictions.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+Cell ontology follows certain conventions for adding taxon constraints. Please do go throug the following documents before adding taxon constraints to CL terms.
```
Comment (bvarner-ebi) [inline docs/taxon-restrictions.md:1]:
```suggestion
The cell ontology (CL) follows certain conventions regarding taxon constraints. 
Please review the following pages before adding taxon constraints to CL terms. 
```

File: docs/taxon-restrictions.md [hunk 2]
```diff
@@ -0,0 +1,7 @@
+Cell ontology follows certain conventions for adding taxon constraints. Please do go throug the following documents before adding taxon constraints to CL terms. 
+
+Explanation of taxon constraints can be found here:
+https://oboacademy.github.io/obook/explanation/taxon-constraints-explainer/ 
+
+A how to guide can be found here:
```
Comment (bvarner-ebi) [inline docs/taxon-restrictions.md:6]:
```suggestion
A how-to guide can be found here:
```

File: mkdocs.yaml [hunk 3]
```diff
@@ -33,5 +33,6 @@ nav:
       - CL Relations: relations_guide.md
       - How to resolve merge conflicts: resolving_merge_conflicts.md
       - Fixing unintended ^^xsd:string diffs: Fixing_xsdstring_diffs.md
+      - Taxon Constraints: taxon-restrictions.md
```
Comment (bvarner-ebi) [inline mkdocs.yaml:36]: Great.
May consider implementing consistent case types in bullets text:

E.g.,  
"Your ODK Repository Overview:" (title case) 
vs
"How to resolve merge conflicts:" (sentence case);

"Imports management:" (sentence case)
vs
"Taxon Constraints" (title case)

**Review by bvarner-ebi — APPROVED** [review]
Great. See inline suggestions.

File: mkdocs.yaml (hunk 3 above)
Comment (shawntanzk) [reply mkdocs.yaml:36]: will fix CL internal docs - the other i *think* needs to be fixed in ODK (think those are shipped with ODK, not sure if they update in the yaml file but I would prefer fixing it there, updating repo with next ODK release, change what needs to be changed

**Pull request merged**